packages/toolpad-components/src/DataGrid.tsx

```diff
@@ -144,6 +144,7 @@ const DataGridComponent = React.forwardRef(function DataGridComponent(
         onColumnOrderChange={handleColumnOrderChange}
         rows={rows}
         columns={columns}
+        getRowId={(row) => row.id ?? crypto.getRandomValues(new Uint16Array(1))[0]}
```

Comment:
```suggestion
```
Let's not make it part of  this PR. I'll leave some comments on the issue you created